Vector/ViewController/RoomMemberDetailsViewController.m

```diff
@@ -0,0 +1,518 @@
+    }
+}
+
+#pragma mark - TableView data source
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    // Check user's power level before allowing an action (kick, ban, ...)
+    MXRoomPowerLevels *powerLevels = [self.mxRoom.state powerLevels];
+    NSUInteger memberPowerLevel = [powerLevels powerLevelOfUserWithUserID:self.mxRoomMember.userId];
+    NSUInteger oneSelfPowerLevel = [powerLevels powerLevelOfUserWithUserID:self.mainSession.myUser.userId];
+    
+    [actionsArray removeAllObjects];
+    
+    // Consider the case of the user himself
+    if ([self.mxRoomMember.userId isEqualToString:self.mainSession.myUser.userId])
+    {
+        [actionsArray addObject:@(MXKRoomMemberDetailsActionLeave)];
+        
+        if (oneSelfPowerLevel >= [powerLevels minimumPowerLevelForSendingEventAsStateEvent:kMXEventTypeStringRoomPowerLevels])
+        {
+            // Check whether the user may reduce his power level to become moderator
+            float normalizedOneSelfPowerLevel = [self.mxRoom.state memberNormalizedPowerLevel:self.mainSession.myUser.userId];
+            if (normalizedOneSelfPowerLevel > 0.5)
+            {
+                [actionsArray addObject:@(MXKRoomMemberDetailsActionSetModerator)];
+            }
+            
+            [actionsArray addObject:@(MXKRoomMemberDetailsActionSetDefaultPowerLevel)];
+            [actionsArray addObject:@(MXKRoomMemberDetailsActionSetCustomPowerLevel)];
+        }
+    }
+    else if (self.mxRoomMember)
+    {
+        // offer to start a new chat only if the room is not a 1:1 room with this user
+        // it does not make sense : it would open the same room
+        MXRoom* room = [self.mainSession privateOneToOneRoomWithUserId:self.mxRoomMember.userId];
+        if (!room || (![room.state.roomId isEqualToString:self.mxRoom.state.roomId]))
+        {
+            [actionsArray addObject:@(MXKRoomMemberDetailsActionStartChat)];
+        }
+        
+        if (self.enableVoipCall)
+        {
+            // Offer voip call options
+            [actionsArray addObject:@(MXKRoomMemberDetailsActionStartVoiceCall)];
+            [actionsArray addObject:@(MXKRoomMemberDetailsActionStartVideoCall)];
+        }
+        
+        // Consider membership of the selected member
+        switch (self.mxRoomMember.membership)
+        {
+            case MXMembershipInvite:
+            case MXMembershipJoin:
+            {
+                // update power level
+                if (oneSelfPowerLevel >= [powerLevels minimumPowerLevelForSendingEventAsStateEvent:kMXEventTypeStringRoomPowerLevels] && oneSelfPowerLevel > memberPowerLevel)
+                {
+                    // Check whether user is admin
+                    float normalizedOneSelfPowerLevel = [self.mxRoom.state memberNormalizedPowerLevel:self.mainSession.myUser.userId];
+                    if (normalizedOneSelfPowerLevel == 1)
```

Comment:
0.5 and 1 normalised values should be replaced by kVectorRoomModeratorLevel and kVectorRoomAdminLevel constant values.